### PR TITLE
Improve docker support

### PR DESCRIPTION
**Pull request by rmg**
* [x] cleanup/refresh docker image
 * [x] a guide for how to manually set up a containerized strong-pm instance
 * [x] a `curl`able script for installing containerized strong-pm as a service

Connected to strongloop-internal/scrum-nodeops#310

### CODE REVIEW

**Comment by rmg** [comment]
Thanks for the proof reading, @sam-github.

The README won't be automatically published to Docker hub, but it certainly can be used there. It's the top-level README.md that is copied over, and that only happens on the first build to set a default, and only if there isn't already content there.

**Comment by sam-github** [comment]
We need at a minimum to be able to specify env vars during the install, so that metrics, auth, and license can be set in the running container. Only AUTH needs to be set at container run time... everything else can be set with `slc pmctl`.

```
# no env install
curl http://strong-pm.io/docker.sh | /bin/sh
# re-install, over an existing install, causing the last install to be replaced
curl http://strong-pm.io/docker.sh STRONGLOOP_AUTH=XXX | /bin/sh
# ... will cause app to be removed... so app(s) will have to be re-deployed... ouch
```

**Comment by rmg** [comment]
@sam-github script is working for me on my Ubuntu and CentOS VMs.

Give it a try!
```
curl -s https://raw.githubusercontent.com/strongloop/strong-pm/improve-docker-support/docker/on-docker.sh | sudo sh -s METRICS=statsd:// AUTH=user:pass
```

**Comment by sam-github** [comment]
Needs some docs. In particular - how to uninstall? I'm struggling with this now... :-(

I can't run `slc pmctl -C http://:8701` after doing this. Just me?

**Comment by sam-github** [comment]
OK, figured out how to uninstall, but not how to connect to it.

**Comment by rmg** [comment]
Connection problem is related to the published version of the strong-pm container using port 7000 instead of 8701.

**Comment by sam-github** [comment]
OK. Well, I don't know how to try it out, then, but I'll assume it will work swimmingly. :-)

The top-level strong-pm README needs a "installing with docker" section (that should mention how to uninstall, too).

Other than that, looks good to me.

**Comment by rmg** [comment]
How did you "uninstall" the service? I'm curious what your definition is and where you drew the line.

**Comment by sam-github** [comment]
I completely failed to kill the service with docker commands, then remembered that upstart was probably keeping it up.

Then I stopped it with `stop strong-pm-container`, and did `sudo rm /etc/init/strong-pm*`

**Comment by rmg** [comment]
Ah, ok, so you didn't actually remove any significant number of bits from disk. I'll add a section to the root README.md that describes how to use the easy installer and links to the detailed docker/README.md for "further reading".

**Comment by rmg** [comment]
@sam-github the strongloop/strong-pm:latest image on docker hub has been updated to the latest and greatest. Running `docker pull strongloop/strong-pm` and then re-running the `curl ... | sudo sh` should now work closer to advertised if you want to try again.

**Comment by rmg** [comment]
@sam-github I've added a section to the top level README.md about installing the container as a service, can you take a look?

File: README.md [hunk 1]
```diff
@@ -86,6 +86,25 @@ to it with `slc deploy http://localhost:8701`.
...
+One of the things you can do with the
+[strongloop/strong-pm](https://registry.hub.docker.com/u/strongloop/strong-pm/)
+image is run it as an OS service without the need to install strong-pm, npm, or
+even node on your server. All you need is Docker!
+
+You can do this [the hard way](docker/README.md) and pull down the image, start
+a container, and write your own Upstart or systemd config file, or you can use
+our install script to do it for you:
+
+    curl -sO https://cdn.rawgit.com/strongloop/strong-pm/master/docker/install.sh | sudo bash
```
Comment (sam-github) [inline README.md]: Is bash required? Can't it be `/bin/sh`? If you found bashisms really handy, that's fine, but its always nice to be real /bin/sh compatible, if possible.

Comment (sam-github) [inline README.md]: And shouldn't it be /bin/bash? Or is bash not in /bin on some systems? Actually, I can see it being in /usr/bin on BSDs. I wonder where solaris falls in all this.

File: docker/install.sh [hunk 2]
```diff
@@ -0,0 +1,138 @@
+#!/bin/sh
```
Comment (sam-github) [inline docker/install.sh]: Seems to be /bin/sh, not bash.

**Comment by sam-github** [comment]
I think there is some inconsistency in your bash vs /bin/sh usage, but other than that, LGTM

**Pull request merged**